tests/scattering1d/test_numpy_backend_1d.py

```diff
@@ -0,0 +1,125 @@
+        x_bad = x.real
+        backend.subsample_fourier(x_bad, 1)
+    assert "should be complex" in te.value.args[0]
+
+def test_pad():
+    N = 128
+    x = np.random.rand(2, 4, N)
+    
+    for pad_left in range(0, N - 16, 16):
+        for pad_right in [pad_left, pad_left + 16]:
+            x_pad = backend.pad(x, pad_left, pad_right)
+            
+            # compare left reflected part of padded array with left side 
+            # of original array
+            for t in range(1, pad_left + 1):
+                assert np.allclose(x_pad[..., pad_left - t], x[..., t])
+            # compare left part of padded array with left side of 
+            # original array
+            for t in range(x.shape[-1]):
+                assert np.allclose(x_pad[..., pad_left + t], x[..., t])
+            # compare right reflected part of padded array with right side
+            # of original array
+            for t in range(1, pad_right + 1):
+                assert np.allclose(x_pad[..., x_pad.shape[-1] - 1 - pad_right + t], x[..., x.shape[-1] - 1 - t])
+            # compare right part of padded array with right side of 
+            # original array
+            for t in range(1, pad_right + 1):
+                assert np.allclose(x_pad[..., x_pad.shape[-1] - 1 - pad_right - t], x[..., x.shape[-1] - 1 - t])
+
+    with pytest.raises(ValueError):
+        backend.pad(x, x.shape[-1], 0)
+
+    with pytest.raises(ValueError):
+        backend.pad(x, 0, x.shape[-1])
+
+def test_unpad():
+    # test unpading of a random tensor
+    x = np.random.rand(8, 4)
+
+    y = backend.unpad(x, 1, 3)
+
+    assert y.shape == (8, 2)
+    assert np.allclose(y, x[:, 1:3])
+
+    N = 128
+    x = np.random.rand(2, 4, N)
+
+    # similar to for loop in pad test
+    for pad_left in range(0, N - 16, 16):
+        pad_right = pad_left + 16
+        x_pad = backend.pad(x, pad_left, pad_right)
+        x_unpadded = backend.unpad(x_pad, pad_left, x_pad.shape[-1] - pad_right)
+        assert np.allclose(x, x_unpadded)
+
+def test_fft_type():
+    x = np.random.rand(8, 4) + 1j * np.random.rand(8, 4)
+
+    with pytest.raises(TypeError) as record:
+        y = backend.rfft(x)
+    assert 'should be real' in record.value.args[0]
+
+    x = np.random.rand(8, 4) 
+
+    with pytest.raises(TypeError) as record:
+        y = backend.ifft(x)
+    assert 'should be complex' in record.value.args[0]
+
+    with pytest.raises(TypeError) as record:
+        y = backend.irfft(x)
+    assert 'should be complex' in record.value.args[0]
+
+def test_fft():
+    x = np.random.randn(2)
+
+    y = np.array([[x[0] + x[1],
+                   x[0] - x[1]]])
+
+    z = backend.rfft(x)
+    assert np.allclose(y, z)
+
+    z_1 = backend.ifft(z)
+    assert np.allclose(x, z_1)
+
+    z_2 = backend.irfft(z)
+    assert not np.iscomplexobj(z_2)
+    assert np.allclose(x, z_2)
+
+    def coefficent(n):
+            return np.exp(-2 * np.pi * 1j * n)
+
+    x_r = np.random.rand(4)
```

Comment:
Why are we doing this twice? Should be enough to just do it below.

Reply:
This is something I was wondering about. Thank you!